### PR TITLE
Make use of brython_done message and css stylesheet injection to make…

### PR DESCRIPTION
**Pull request by iansparks**
… loader less intrusive

@rayluo this PR uses 2 new techniques to make the loader less intrusive. It now only requires you to add the loader.js file and then call start_loader(). The loader.js file injects the css style tag into the current document so that you don't have to have the loader.css file. Secondly, the loader.js file adds a listener for the "brython_done" event which brython itself posts when its all done. This triggers the loader to remove its HTML and the injected style sheet. 

You *might* feel this is too much magic but its less boilerplate to remember....

### CODE REVIEW

**Review by rayluo — APPROVED** [review]
Nice magic! Love it!

Since we went this far, I go ahead to remove the no-longer-necessary `div#main`, and also group the non-boilerplate code together. What do you think, @iansparks ?

**Comment by iansparks** [comment]
Works for me @rayluo though I was puzzled that main.py was loaded before the brython files - until I remembered, *yet again* that javascript goes before python. I also think the README may need a bit of updating.

File: website/index.html [hunk 1]
```diff
@@ -29,24 +24,32 @@ <h2>Echo</h2>
...
-      <!-- The real project content ends from here -->
-    </div>
+      <!--
+        Your main logic should go into main.py.
+        Parsing this code may cause a delay but the css animations from load.js
+        should continue to run.
 
+        It is important to remember that all text/python scripts are loaded and run
+        in order AFTER all the javascript scripts.
+        You cannot interleave python/javascript scripts and expect them to run in order.
```
Comment (rayluo) [inline website/index.html:34]: @iansparks , to address your latest comment, I move this part of "important message" into right before the inclusion of `main.py`.

File: README.md [hunk 2]
```diff
@@ -37,7 +37,17 @@ You can use this repo as a template to start your own project.
    and then visit `http://localhost:8000` to see it in action.
 
 That is all. Once you are familiar with this project's structure,
-you can modify it by changing
-the content inside `<div id="main" ...>...</div>` from `index.html`
+you can modify it by changing the content between
+`<!-- The real project content starts from here -->` and `<!-- The real project content ends here -->`
+in `index.html`,
```
Comment (rayluo) [inline README.md:42]: @iansparks , I believe this is what you mean by "the README need a bit of updating"

File: README.md [hunk 3]
```diff
@@ -37,7 +37,17 @@ You can use this repo as a template to start your own project.
...
 and content inside `main.py`.
 
+Alternatively, if you already have an existing project,
+you can add the loading animation by including this snippet into your `index.html`
+
+    ```html
+    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">
+
+    <script src="https://rayluo.github.io/brython-project-template/loader.js"></script>
```
Comment (rayluo) [inline README.md:51]: And, thanks to your recent work, now the modular loading animation could be used in this way (but that link would only be available AFTER this pr is merged).

**Review by rayluo — COMMENTED** [review]
@iansparks , how about the following changes?

File: README.md (hunk 2 above)
Comment (iansparks) [reply README.md:42]: It says 

```
...Once you are familiar with this project's structure, you can modify it by changing the content inside <div id="main" ...>...</div> from index.html and content inside main.py."
```
But you remove id="main" from the div?

File: README.md (hunk 3 above)
Comment (iansparks) [reply README.md:51]: Yes, nice thinking. Sadly still need fontawesome if you want to make use of those icons. We could make more flexible still but I think useful as-is.

File: README.md (hunk 2 above)
Comment (rayluo) [reply README.md:42]: Yes, and this latest commit removed that line. I know you said you are no longer a frequent github user nowadays :-)  but if you are reading [this PR from the gitub web ui](https://github.com/rayluo/brython-project-template/pull/3/files), it shows the deleted lines in red.

So, I guess you are satisfied with these finishing touch on this PR?

Comment (iansparks) [reply README.md:42]: Yes, go for it.

**Pull request merged**